File: Sully/src/sully.c
#include <stdio.h>
#include <fcntl.h>
#include <string.h>
#include <stdlib.h>
#include <unistd.h>
int main()
{
int i = 5;
if(i <= 0)return(0);if(getenv("child"))i--;
char *filename = NULL;
asprintf(&filename, "Sully_%d.c", i);
int fd = open(filename, O_WRONLY | O_CREAT | O_TRUNC, S_IWUSR | S_IRUSR);
char*str="#include <stdio.h>%c#include <fcntl.h>%c#include <string.h>%c#include <stdlib.h>%c#include <unistd.h>%cint main()%c{%cint i = %d;%cif(i <= 0)return(0);if(getenv(%cchild%c))i--;%cchar *filename = NULL;%casprintf(&filename, %cSully_%%d.c%c, i);%cint fd = open(filename, O_WRONLY | O_CREAT | O_TRUNC, S_IWUSR | S_IRUSR);%cchar*str=%c%s%c;%cdprintf(fd, str, 10, 10, 10, 10, 10, 10, 10, i, 10, 34, 34, 10, 10, 34, 34, 10, 10, 34, str, 34, 10, 10, 34, 34, 10, 34, 34, 10, 10, 10);%cchar *exec;asprintf(&exec, %c/usr/bin/gcc %%s -o Sully_%%d%c, filename, i);system(exec);%casprintf(&exec, %cenv -i child=1 ./Sully_%%d%c, i);if(i>0)system(exec);%creturn (0);%c}%c";
dprintf(fd, str, 10, 10, 10, 10, 10, 10, 10, i, 10, 34, 34, 10, 10, 34, 34, 10, 10, 34, str, 34, 10, 10, 34, 34, 10, 34, 34, 10, 10, 10);
char *exec;asprintf(&exec, "/usr/bin/gcc %s -o Sully_%d", filename, i);system(exec);
asprintf(&exec, "env -i child=1 ./Sully_%d", i);if(i>0)system(exec);
return (0);
}

File: Bonus/Sully/src/sully.py
#!/usr/bin/python3
import os;
if __name__ == '__main__':
    i = 5;
    if (os.environ.get('child') != None):
        i = i - 1;
    filename = 'Sully_%d.py' % (i);
    str="#!/usr/bin/python3%cimport os;%cif __name__ == '__main__':%c    i = %d;%c    if (os.environ.get('child') != None):%c        i = i - 1;%c    filename = 'Sully_%%d.py' %% (i);%c    str=%c%s%c;%c    with open(filename, 'w') as fd:%c        fd.write(str %% (10, 10, 10, i, 10, 10, 10, 10, 34, str, 34, 10, 10, 10, 10, 10));%c    if (i > 0):%c        os.system('env -i child=1 /usr/bin/python3 ./%%s' %% (filename));%c";
    with open(filename, 'w') as fd:
        fd.write(str % (10, 10, 10, i, 10, 10, 10, 10, 34, str, 34, 10, 10, 10, 10, 10));
    if (i > 0):
        os.system('env -i child=1 /usr/bin/python3 ./%s' % (filename));
